pytorch_lightning/utilities/debug_examples.py

```diff
@@ -0,0 +1,300 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
Open to different filename suggestions or file locations.

Reply:
yep feel like the utils shall be completely inter-dependency free...
hoe about `PL/develop/samples`?

Reply:
we also have a utilities file `debugging.py` so my suggestion is to move it there or adjust naming to debugging_examples :)

Reply:
> hoe about PL/develop/samples?

Are you saying `pytorch-lightning/pytorch_lightning/develop/samples` or `pytorch-lightning/develop/samples`?

The latter wouldn't work as we need it to be distributed with the source.

> debugging_examples

That's fine with me

Reply:
What about `pytorch_lightning/utilities/examples/...` ?

Reply:
IMO we want to keep this list of in-source examples as small as possible. Making it a directory might not encourage that